10-ConcurrencyAndSynchronization/my_thread.c

```diff
@@ -0,0 +1,72 @@
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/kthread.h>
+#include <linux/sched.h>
+#include <linux/time.h>
+#include <linux/timer.h>
+#include <linux/delay.h>
+#include <linux/spinlock.h>
+#include <linux/rtmutex.h>
+#include <linux/semaphore.h>
+
+#define MAX_THREADS 4
+
+static struct task_struct *threads[MAX_THREADS];
+
+//Please use one single RT_MUTEX or SPINLOCK or SEMAPHORE
+static DEFINE_RT_MUTEX(mutexname);
+// static DEFINE_SPINLOCK(spinlockname);
```

Comment:
try to create commit history instead of commented code

Reply:
Sorry,  but I made this way to have these examples in my code for future uses.
I don't wish to used different commits for different thread implementation approaches.
Please let me leave this as is to have examples on single IDE page in my branch.